src/Data/Loader/UserLoader.php

```diff
@@ -61,16 +68,78 @@ public function loadKeys( array $keys ) {
 		if ( empty( $users ) || ! is_array( $users ) ) {
 			return [];
 		}
+		
+		$this->load_published_author_ids( $keys );
+		
+		foreach ( $keys as $id ) {
+			$user = get_user_by( 'id', $id );
+
+			$user_has_published_posts = in_array(
+				absint( $id ),
+				$this->published_authors,
+				true 
+			);
+
+			$user->has_published_posts = $user_has_published_posts;
 
-		/**
-		 *
-		 */
-		foreach ( $keys as $key ) {
-			$user                   = get_user_by( 'id', $key );
 			$all_users[ $user->ID ] = new User( $user );
 		}
+
 		return $all_users;
 
 	}
 
+	/**
+	 * This method accepts an array of user ID's, and stores an array of 
+	 * user ID's for the subset that are published authors.
+	 * 
+	 * @param array $keys An array of post ID's
+	 */
+	protected function load_published_author_ids( $ids ) {
+		$post_types = get_post_types( [ 'show_in_graphql' => true ] );
+
+        unset( $post_types[ 'attachment' ] );
+        unset( $post_types[ 'revision' ] );
+		
+		$stringified_ids        = implode( ',', $ids );
+		$stringified_post_types = '"' . implode( '", "', $post_types ) . '"';
+
+		$where = get_posts_by_author_sql( $post_types, true, null, false );
+		
+		global $wpdb;
+        $results = $wpdb->get_results( 
```

Comment:
Spacing looks funky here. 

We'll also need to make sure to use `$wpdb->prefix` for the post tables as not all WP installs use `wp_` as the prefix, so we can't hardcode that.

Reply:
Just pushed up a fix for this.